src/datadog/platform_util.cpp

```diff
@@ -189,6 +189,26 @@ int get_process_id() {
 #endif
 }
 
+std::string get_process_name() {
+#if defined(__APPLE__) || defined(__FreeBSD__)
+  return getprogname();
```

Comment:
```suggestion
  char* process_name = getprogname();
  return (process_name != nullptr) ? process_name : "unknown-service";
```

`man getprogname`
> The getprogname() function returns the name of  the  program.   If  the
       name has	not been set yet, it will return NULL.

An `std::string` constructed with a NULL pointer will throw an exception.